CesiumGeospatial/src/EarthGravitationalModel96Grid.cpp

```diff
@@ -0,0 +1,119 @@
+#include <CesiumGeospatial/EarthGravitationalModel96Grid.h>
+#include <CesiumGeospatial/Ellipsoid.h>
+#include <CesiumUtility/Math.h>
+
+#include <algorithm>
+#include <fstream>
+#include <vector>
+
+using namespace CesiumUtility;
+
+namespace CesiumGeospatial {
+
+// WW15MGH.DAC has 721 rows representing the range (90N, 90S) and 1440 columns
+// representing the range (0E, 360E)
+
+// The number of rows in the file
+const int NUM_ROWS = 721;
+// The number of columns in the file
+const int NUM_COLUMNS = 1440;
+
+std::optional<EarthGravitationalModel96Grid>
+CesiumGeospatial::EarthGravitationalModel96Grid::fromFile(
+    const std::string& filename) {
+  std::ifstream file(filename, std::ios::binary | std::ios::ate);
+  if (!file.good()) {
+    return std::nullopt;
+  }
+
+  std::streamsize size = file.tellg();
+  file.seekg(0, std::ios::beg);
+
+  std::vector<std::byte> buffer(static_cast<size_t>(size));
+  file.read(reinterpret_cast<char*>(buffer.data()), size);
+  return fromBuffer(buffer);
+}
+
+std::optional<EarthGravitationalModel96Grid>
+CesiumGeospatial::EarthGravitationalModel96Grid::fromBuffer(
+    const gsl::span<const std::byte>& buffer) {
+  const int expectedBytes = NUM_ROWS * NUM_COLUMNS * sizeof(int16_t);
+
+  // Not enough data - is this a valid WW15MGH.DAC?
+  if (buffer.size_bytes() < expectedBytes) {
+    return std::nullopt;
+  }
+
+  std::vector<int16_t> gridValues;
```

Comment:
Use `gridValues.reserve` to avoid reallocating the array multiple times while adding values.